flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/operators/windowing/StreamDiscretizer.java

```diff
@@ -183,6 +173,16 @@ public void open(org.apache.flink.configuration.Configuration parameters) throws
 		}
 	}
 
+	@Override
+	public void close() throws Exception {
+		super.close();
+		if (activePolicyThread != null) {
+			activePolicyThread.interrupt();
+		}
+
+		emitWindow();
```

Comment:
I think it's fine in streaming. open(), receiveElement(), close() are all called in the inner loop of StreamTask.invoke()